### DISCUSSION TITLE
SyntaxError: Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by vixeven** [vercel/next.js#15957] [Help]
I tried to integrate [Fluid Player](https://docs.fluidplayer.com/docs/integration/react/) with latest version of Next.js.

https://codesandbox.io/s/fervent-montalcini-i690v?file=/pages/index.tsx

```
import './polyfills';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Module._compile (internal/modules/cjs/loader.js:891:18)
```

Can someone explain why this is happening? I tried `dynamic` import with `{ ssr: false }` but same error occurred. 
Any help or insight would be greatly appreciated.

**Comment by vasconx** [comment]
Okay, I downloaded it here to see the problem. What he really expects is the require syntax in this part of the code for some reason. The little I've been studying has to do with the way the transpiler reads the code, someone with more property will be able to correct me, but for some reason for this part of the package the program expects to receive the require syntax.

**Comment by vasconx** [comment]
So, what I did to run, I just changed the imports and then it will tell you that windows is not defined because it is running on the server side, so the 'dynamic' you used is good tone and the player works, but I think that it can be palliative, sometimes it changed something that needed directly in node_modules but if there is a minified version (example.min.js) when the deploy occurs, the min version wins.

**Comment by vasconx** [comment]
my-app/node_modules/fluid-player/src/index.js:
..
require('./polyfills');
const fluidPlayerInitializer = require('./fluidplayer.js');

module.exports = fluidPlayerInitializer;
...

**Comment by Addvilz** [comment]
I am not familiar with next.js, but it looks like you are trying to load Fluid Player using CommonJS loader. Fluid Player is a ES2015 module - so it won't work. Besides, Fluid Player uses dynamic imports as well, and those will break if you try above (hls, dash support etc).

Closing https://github.com/fluid-player/fluid-player/issues/506 since this does not seem to be a Fluid Player issue.

**Comment by vixeven** [comment]
So, what will be the fix for this integration? How should I import this package?
Next.js does not support ES2015 modules? When I run `yarn dev` command, property `module` is reseted to value **`esnext`**.

@nastari @timneutkens
